src/pages/workspace/reimburse/WorkspaceRateAndUnitPage.js

```diff
@@ -39,82 +36,34 @@ const defaultProps = {
+function WorkspaceRateAndUnitPage(props) {
+    const fetchData = useCallback(() => {
+        if (lodashGet(props, 'policy.customUnits', []).length !== 0) {
             return;
         }
-        // When this page is accessed directly from url, the policy.customUnits data won't be available,
-        // and we should trigger Policy.openWorkspaceReimburseView to get the data
 
         BankAccounts.setReimbursementAccountLoading(true);
-        Policy.openWorkspaceReimburseView(this.props.policy.id);
-    }
-
-    componentDidUpdate(prevProps) {
-        // We should update rate input when rate data is fetched
-        if (prevProps.reimbursementAccount.isLoading === this.props.reimbursementAccount.isLoading) {
-            return;
-        }
+        Policy.openWorkspaceReimburseView(props.policy.id);
+    }, [props]);
 
-        this.resetRateAndUnit();
-    }
+    useEffect(() => {
+        fetchData();
+    }, [fetchData]);
 
-    getUnitItems() {
-        return [
-            {label: this.props.translate('common.kilometers'), value: CONST.CUSTOM_UNITS.DISTANCE_UNIT_KILOMETERS},
-            {label: this.props.translate('common.miles'), value: CONST.CUSTOM_UNITS.DISTANCE_UNIT_MILES},
-        ];
-    }
+    const getUnitItems = () => [
+        {label: props.translate('common.kilometers'), value: CONST.CUSTOM_UNITS.DISTANCE_UNIT_KILOMETERS},
+        {label: props.translate('common.miles'), value: CONST.CUSTOM_UNITS.DISTANCE_UNIT_MILES},
+    ];
 
-    getRateDisplayValue(value) {
-        const numValue = this.getNumericValue(value);
-        if (Number.isNaN(numValue)) {
-            return '';
-        }
-        return numValue.toString().replace('.', this.props.toLocaleDigit('.')).substring(0, value.length);
-    }
-
-    getNumericValue(value) {
-        const numValue = NumberUtils.parseFloatAnyLocale(value.toString());
-        if (Number.isNaN(numValue)) {
-            return NaN;
-        }
-        return numValue.toFixed(3);
-    }
-
-    resetRateAndUnit() {
-        const distanceCustomUnit = _.find(lodashGet(this.props, 'policy.customUnits', {}), (unit) => unit.name === CONST.CUSTOM_UNITS.NAME_DISTANCE);
-        const distanceCustomRate = _.find(lodashGet(distanceCustomUnit, 'rates', {}), (rate) => rate.name === CONST.CUSTOM_UNITS.DEFAULT_RATE);
-
-        this.setState({
-            rate: PolicyUtils.getUnitRateValue(distanceCustomRate, this.props.toLocaleDigit),
-            unit: lodashGet(distanceCustomUnit, 'attributes.unit', CONST.CUSTOM_UNITS.DISTANCE_UNIT_MILES),
-        });
-    }
-
-    saveUnitAndRate(unit, rate) {
-        const distanceCustomUnit = _.find(lodashGet(this.props, 'policy.customUnits', {}), (u) => u.name === CONST.CUSTOM_UNITS.NAME_DISTANCE);
+    const saveUnitAndRate = (unit, rate) => {
+        const distanceCustomUnit = _.find(lodashGet(props, 'policy.customUnits', {}), (u) => u.name === CONST.CUSTOM_UNITS.NAME_DISTANCE);
```

Comment:
NAB not your code but I think this is more readable:
```suggestion
        const distanceCustomUnit = _.find(lodashGet(props, 'policy.customUnits', {}), (customUnit) => customUnit.name === CONST.CUSTOM_UNITS.NAME_DISTANCE);
```